src/LiveComponent/assets/src/Component/index.ts

```diff
@@ -300,15 +300,59 @@ export default class Component {
 
         this.backendRequest.promise.then(async (response) => {
             const backendResponse = new BackendResponse(response);
-            const html = await backendResponse.getBody();
 
             // clear sent files inputs
             for (const input of Object.values(this.pendingFiles)) {
                 input.value = '';
             }
 
-            // if the response does not contain a component, render as an error
             const headers = backendResponse.response.headers;
+            if (headers.get('X-Live-Download')) {
+                const headerContentDisposition = headers.get('Content-Disposition');
+                if (
+                    !headerContentDisposition
+                    || !(headerContentDisposition?.includes('attachment') || headerContentDisposition?.includes('inline')) 
+                    || !headerContentDisposition?.includes('filename=')
+                ) {
+                    throw new Error('Invalid LiveDownload response');
+                }
+
+                const fileSize = Number.parseInt(headers.get('Content-Length') || '0');
+                if (fileSize > 10000000) {
+                    throw new Error('File is too large to download (10MB limit)');
+                }
+
+                const fileName = headerContentDisposition.split('filename=')[1];
+                if (!fileName) {
+                    throw new Error('No filename found in Content-Disposition header');
+                }
+
+                const blob = await backendResponse.getBlob();
```

Comment:
Does this load the entire file (up to our max) into memory?

Reply:
It depends... if you trigger the "click" and the file start downloading I think browser play the "pipe" role here.

But during my test I managed to crashed pretty violently Chrome and Safari multiple times.